frontend/src/Sidebar.tsx

```diff
@@ -1,13 +1,103 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { Link } from "react-router-dom";
+
+import styled from "styled-components";
+
 import "./Sidebar.css";
+
 import { EditMode } from "./Crossword";
 
 interface SidebarProps {
   mode: EditMode;
   setMode: (arg: EditMode) => void;
+  brushSize: number;
+  setBrushSize: (arg: number) => void;
+}
+
+const InputRangeWithLabelDiv = styled.div`
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
+    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
+    sans-serif;
+  margin: 0 16px;
+  label {
+    color: var(--dark-gray);
+    font-weight: 600;
+    cursor: pointer;
+  }
+`;
+
+const InputWrapper = styled.div`
+  display: flex;
+  margin-top: 5px;
+  input[type="number"] {
+    width: 25%;
+    min-width: 40px;
+  }
+  input[type="range"] {
+    width: 50%;
+    margin: auto;
+  }
+`;
+
+interface RangeWithLabelProps {
+  label: string;
+  value: number;
+  updateValue: (arg: number) => void;
+  min?: number;
+  max?: number;
 }
 
+const InputRangeWithLabel: React.FC<RangeWithLabelProps> = (props) => {
+  const [textInputValue, setTextInputValue] = useState(props.value);
+
+  useEffect(() => {
+    setTextInputValue(props.value);
+  }, [props.value]);
+
+  const min = Math.floor(props.min || 0);
+  const max = Math.ceil(props.max || 8);
+  const stepSize = props.value <= 1 ? 0.25 : props.value <= 2.5 ? 0.5 : 1;
+
+  const updateValue = (val: string) => {
+    props.updateValue(Math.max(min, parseFloat(val)));
+  };
+
+  return (
+    <InputRangeWithLabelDiv>
+      <label>
+        {props.label}
+        <InputWrapper>
+          <input
```

Comment:
I would suggest only having the slider here. Maybe the up-and-down buttons but not the input field. It is also a bit buggy at the moment, e.g.:
- current value is 0.5, want 2
- try deleting with backspace -> resets value to 0.25

Only having the range would also make the logic simpler, no need for any state in this component etc.